smartsim/_core/mli/infrastructure/storage/featurestore.py

```diff
@@ -57,6 +81,22 @@ class FeatureStore(ABC):
     """Abstract base class providing the common interface for retrieving
     values from a feature store implementation"""
 
+    def __init__(self) -> None:
+        self._reserved_write_enabled = False
+
+    def _check_reserved(self, key: str) -> None:
+        """A utility method used to verify access to write to a reserved key
+        in the FeatureStore. Used by subclasses in __setitem___ implementations
+
+        :param key: a key to compare to the reserved keys
+        :raises SmartSimError: if the key is reserved"""
+        reserved_key_match = ReservedKeys.from_string(key)
+        if reserved_key_match and not self._reserved_write_enabled:
```

Comment:
Your sample combines into the same statement in reversed order, but that order does let me avoid the constructor exception, e.g. `return cls("not-in-there")` so I think it's probably the better route to avoid unnecessary exceptions being raised. Good call.